src/TextField/TextFieldLabel.spec.js

```diff
@@ -41,4 +41,14 @@ describe('<TextFieldLabel>', () => {
     const wrapper = shallow(<TextFieldLabel shrink />);
     assert.strictEqual(wrapper.hasClass(classes.shrink), true, 'should have the shrink class');
   });
+
+  it('should show an asterisk if required is set', () => {
+    const wrapper = shallow(<TextFieldLabel required />);
+    assert.strictEqual(wrapper.text().indexOf('*') >= 0, true, 'should show an asterisk');
+  });
+
+  it('should not show an asterisk by default', () => {
+    const wrapper = shallow(<TextFieldLabel />);
+    assert.strictEqual(wrapper.text().indexOf('*') >= 0, false, 'should not show an asterisk');
```

Comment:
Whoops, sorry didn't see your answer and just removed my comment after looking for it myself.